File: dotnet/samples/SimpleOrm.Sample/Models/Tables/Role.cs
namespace SimpleOrm.Sample.Models;

/// <summary>Table <c>roles</c> (STRICT). Key: <c>id</c>, database-generated.</summary>
[Table("roles")]
[Index(nameof(Name), Unique = true)]
public sealed class Role : BaseModel
{
    [Key]
    [Generated]
    [Column]
    public long Id { get; set; }

    /// <summary>Column renamed to <c>role_name</c> by migration V0004.</summary>
    [Column("role_name")]
    public required string Name { get; set; }

    /// <summary>The reverse side of <see cref="User.Roles"/>, through the same link (ADR-0019).</summary>
    [ManyToMany(typeof(UserRole))]
    public IReadOnlyList<User> Users { get; private set; } = [];
}

File: php/samples/SimpleOrm.Sample/src/Models/Tables/Role.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Sample\Models\Tables;

use SimpleOrm\Metadata\Attributes\Column;
use SimpleOrm\Metadata\Attributes\Generated;
use SimpleOrm\Metadata\Attributes\Index;
use SimpleOrm\Metadata\Attributes\Key;
use SimpleOrm\Metadata\Attributes\ManyToMany;
use SimpleOrm\Metadata\Attributes\Table;

/** Table `roles` (STRICT). Key: `id`, database-generated. */
#[Table('roles')]
#[Index(['name'], unique: true)]
final class Role extends BaseModel
{
    #[Key]
    #[Generated]
    #[Column]
    public int $id;

    /** Column renamed to `role_name` by migration V0004. */
    #[Column('role_name')]
    public string $name;

    /** The reverse side of `User::$roles`, through the same link (ADR-0019). Declaration-only in this port. */
    #[ManyToMany(User::class, through: UserRole::class)]
    public private(set) array $users = [];
}
